client/signup/steps/user/index.jsx

```diff
@@ -73,7 +74,17 @@ export class UserStep extends Component {
 	}
 
 	componentWillMount() {
+		const { oauth2Signup, initialContext } = this.props,
+			clientId =
+				get( initialContext, [ 'query', 'oauth2_client_id' ] )
```

Comment:
You can pass default value, so
`clientId = get( initialContext, 'query.oauth2_client_id', null );` is enough imo.

[Another example](https://github.com/Automattic/wp-calypso/blob/5f15f3365213253724ca0d8798c6cb8160f00bc9/client/extensions/woocommerce/state/sites/products/selectors.js#L42)

Reply:
You're right of course! Not sure where I saw that other syntax. I've updated it as this is much clearer.